webapp/src/components/post_type/action_view/action_view.jsx

```diff
@@ -72,36 +76,24 @@ export default class ActionView extends React.PureComponent {
         actions.
             filter((action) => action.id && action.name).
             forEach((action) => {
-                switch (action.type) {
-                case ActionButtonType.BUTTON:
-                    if (this.isAddOptionAction(action) && !this.hasPermissionForAddOption(metadata)) {
-                        // skip to add the button for addOption if the user doesn't have permission for adding options
-                        break;
-                    }
-                    content.push(
-                        <ActionButton
-                            key={action.id}
-                            action={action}
-                            postId={this.props.post.id}
-                            hasVoted={this.hasVoted(action, metadata)}
-                        />,
-                    );
-                    break;
-                case ActionButtonType.MATTERPOLL_ADMIN_BUTTON:
-                    if (metadata.can_manage_poll) {
-                        adminContent.push(
-                            <ActionButton
-                                key={action.id}
-                                action={action}
-                                postId={this.props.post.id}
-                                hasVoted={false}
-                            />,
-                        );
-                    }
-                    break;
-                default:
-                    break;
+                if (action.type !== ActionButtonType.BUTTON) {
+                    return;
+                }
+                if (this.isAddOptionAction(action) && !this.hasPermissionForAddOption(metadata)) {
+                    // skip to add the button for addOption if the user doesn't have permission for adding options
+                    return;
+                }
+                if (this.isPollManagementAction(action) && !metadata.can_manage_poll) {
+                    return;
                 }
+                content.push(
+                    <ActionButton
+                        key={action.id}
+                        action={action}
+                        postId={this.props.post.id}
+                        hasVoted={this.hasVoted(action, metadata)}
```

Comment:
Do we need the `hasVoted` props for management actions?

Reply:
As you said, `hasVoted` props doesn't needed for management actions, but this line is also used for answer options, so this line is needed.